tikv/raftstore/read.go

```diff
@@ -0,0 +1,71 @@
+package raftstore
+
+import (
+	stdatomic "sync/atomic"
+	"time"
+	"unsafe"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/raft_cmdpb"
+	"github.com/uber-go/atomic"
+)
+
+type LeaseChecker interface {
+	CheckLease(req *raft_cmdpb.RaftCmdRequest, snapTime time.Time) (bool, error)
+}
+
+type leaseChecker struct {
+	invalid          atomic.Bool
+	peerID           atomic.Uint64
+	term             atomic.Uint64
+	appliedIndexTerm atomic.Uint64
+	leaderLease      unsafe.Pointer // *RemoteLease
+	region           unsafe.Pointer // *metapb.Region
+}
+
+func (c *leaseChecker) CheckLease(req *raft_cmdpb.RaftCmdRequest, snapTime time.Time) (bool, error) {
```

Comment:
I  think it would be better to use `*kvrpcpb.Context` than `*raft_cmdpb.RaftCmdRequest`.
The caller only calls this method in read-requests, it doesn't have to compose a  `RaftCmdRequest`.

Reply:
If lease expired, there should be a way to let the caller wait for the lease to update.